sr_utilities_common/scripts/sr_utilities_common/conditional_delayed_rostool.py

```diff
@@ -27,13 +27,22 @@ def __init__(self, element_type, required_elements_list):
         self._required_elements_list = required_elements_list
         self.missing_elements = []
 
-    def execute(self):
+    def check_if_required_element_is_published(self):
         all_elements_available = False
         roscore_published_elements = self._retrieve_available_elements(self._element_type)
-        self._check_if_element_is_available(roscore_published_elements, self._required_elements_list)
-        if 0 == len(self._required_elements_list):
+        if self._required_elements_list:
+            for element in self._required_elements_list:
+                if element:
+                    if any(element in sublist for sublist in roscore_published_elements):
+                        rospy.loginfo("Found %s", element)
+                        self._required_elements_list.remove(element)
```

Comment:
you shouldnt remove elements from list while iterating over it:
https://stackoverflow.com/questions/6260089/strange-result-when-removing-item-from-a-list